### ISSUE TITLE
Cannot configure mongodb connection - from code

### ISSUE THREAD

**Issue by Glagnar** [node-red/node-red-nodes#163]
I would like to dynamically configure the mongodb node, so it can connect to different servers. 
I dont mind having a configuration file for it, but I want it separated form the sheet's json file.

**Comment by dceejay** [comment]
How dynamically ? just at startup (as is implied by your mention of a config file) ? 
If just at startup then you can use  $(env-variable-name)  in most config fields to pick up an environment variable that you could preset before starting Node-RED.

**Comment by Glagnar** [comment]
At startup would be fine.
So, to be more concrete - if I want to set the IP address or port of the mongoldb server, which variable do I need to set. Do I set it in the 'functionGlobalContext' and then expect it to 'just-work'?

**Comment by dceejay** [comment]
You can set an environment variable outside of Node-RED - how that is done depends on your operating system. eg Linux - export MONGO_SERVER=192.168.1.20    or  Windows http://www.computerhope.com/issues/ch000549.htm 
and then start Node-RED.   Inside the config box for mongo you can then
![image](https://cloud.githubusercontent.com/assets/5375409/12320173/1bfaee28-ba9f-11e5-8805-49c6e228d7d6.png)

**Comment by Glagnar** [comment]
Fantastic! I didn't know this.
Thank you very much.

**Comment by dceejay** [comment]
Great - 
(we have the node-red google group for general usage questions - https://groups.google.com/forum//#!forum/node-red )

**Comment by Glagnar** [comment]
Thanks. I have joined the group, and using the OS' environment variables works fine.
Before I start reworking everything I have written to use environment variables, perhaps I could ask you if there was a way to use the nodered context instead.
The case is that I load a lot of other configuration from a JSON into functionGlobalContext. This means that in a node, I can reference something like "context.global.mongoDatabase.host;". I tried $(context.global.hosts.mongoDatabase.host) but this of course does not work.
